src/core/main.c

```diff
@@ -294,6 +295,25 @@ static int parse_crash_chvt(const char *value) {
         return 0;
 }
 
+static int parse_confirm_spawn(const char *value, char **console) {
+        char *cons;
+        int r;
+
+        r = parse_boolean(value);
+        if (r == 0) {
+                *console = NULL;
+                return 0;
+        }
+        if (r > 0)
+                value = "/dev/console";
```

Comment:
`*console` might be set multiple times (if the user species the same parameter multiple times). So `*console` might point to previously allocated string which should be freed. Thus, please use `free_and_strdup` to deallocate the old contents.

```
r = parse_boolean(value);
if (r == 0)
   value = NULL;
else if (r > 0)
  value = "/dev/console";
return free_and_strdup(console, value);
```